components/automate-cli/cmd/chef-automate/verify.go

```diff
@@ -204,126 +197,286 @@ func (v *verifySystemdCreateFlow) runVerifySystemdCreateCmd(cmd *cobra.Command,
+	if err != nil {
+		return fmt.Errorf("error while doing batch-check API call for %s: %v", nodeType, err)
+	} else {
+		defer response.Body.Close()
+		resultBytes, err := v.getResultFromResponseBody(response.Body)
+		if err != nil {
+			return err
 		}
-		return status.New(status.InvalidCommandArgsError, errProvisonInfra)
+		v.Writer.Printf("Response for batch-check API on %s: \n%s\n", nodeType, string(resultBytes))
 	}
-	return status.New(status.InvalidCommandArgsError, "Managed Services flag is not set. Cannot verify the config.")
+	v.Writer.Printf("Batch-check API call for %s completed\n", nodeType)
+	return nil
 }
 
-func (v *verifyCmdFlow) verifyStandaloneDeploy(configPath string) error {
-	if v.A2HARBFileExist {
-		return status.New(status.InvalidCommandArgsError, "Deployment type does not match with the requested flag.")
+// Copies CLI binary to remote nodes
+func (v *verifyCmdFlow) copyCLIOnRemoteNodes(destFileName string, sshConfig sshutils.SSHConfig, hostIPs []string) error {
+	v.Writer.Println("Copying automate-verify CLI to remote nodes")
+	currentBinaryPath, err := v.SystemdCreateUtils.GetBinaryPath()
+	if err != nil {
+		return err
 	}
-	if err := v.Verification.VerifyStandaloneDeployment(configPath); err != nil {
-		return status.Annotate(err, status.ConfigError)
+	copyResults := v.SSHUtil.CopyFileToRemoteConcurrently(sshConfig, currentBinaryPath, destFileName, false, hostIPs)
+	isError := false
+	for _, result := range copyResults {
+		if result.Error != nil {
+			v.Writer.Errorf("Remote copying automate-verify CLI failed on node : %s with error: %v\n", result.HostIP, result.Error)
+			isError = true
+		}
 	}
-
+	if isError {
+		return fmt.Errorf("remote copying failed")
+	}
+	v.Writer.Println("Copying automate-verify CLI to remote nodes completed")
 	return nil
 }
 
-func (v *verifyCmdFlow) verifyHaOnpremDeploy(configPath string) error {
-	if err := v.Verification.VerifyOnPremDeployment(configPath); err != nil {
-		return status.Annotate(err, status.ConfigError)
+// Starts automate-verify service on remote nodes
+func (v *verifyCmdFlow) StartServiceOnRemoteNodes(destFileName string, sshConfig sshutils.SSHConfig, hostIPs []string) error {
+	v.Writer.Println("Creating automate-verify service on remote nodes")
+	cmd := fmt.Sprintf("sudo /tmp/%s verify systemd-service create", destFileName)
+	excuteResults := v.SSHUtil.ExecuteConcurrently(sshConfig, cmd, hostIPs)
+	isError := false
+	for _, result := range excuteResults {
+		if result.Error != nil {
+			v.Writer.Errorf("Remote execution of cmd '%s' automate-verify CLI failed on node : %s with error: %v\n", cmd, result.HostIP, result.Error)
+			isError = true
+		}
 	}
+	if isError {
+		return fmt.Errorf("remote execution failed")
+	}
+	// Wait for given seconds for the service to start
+	time.Sleep(SLEEP_DURATION)
+	v.Writer.Println("Creating automate-verify service on remote nodes completed")
```

Comment:
Can we have "Created automate-verify service on remote nodes"

Reply:
done